dart/common/StlHelpers.h

```diff
@@ -41,9 +41,26 @@
 #include <cstddef>
 #include <vector>
 
+namespace std {
+
+//==============================================================================
+template<typename T, typename... Args>
+unique_ptr<T> make_unique(Args&&... args)
+{
+  return unique_ptr<T>(new T(forward<Args>(args)...));
+}
+// TODO(JS): This is a stopgap solution as it was omitted from C++11 as "partly
+// an oversight". This can be replaced by std::make_unique<T> of the standard
+// library when we migrate to using C++14.
+
+} // namespace std
```

Comment:
It is undefined to add declarations or definitions to the `std` namespace:

> `[C++11: 17.6.4.2.1/1]:` The behavior of a C++ program is undefined if it adds declarations or definitions to namespace `std` or to a namespace within namespace `std` unless otherwise specified. A program may add a template specialization for any standard library template to namespace `std` only if the declaration depends on a user-defined type and the specialization meets the standard library requirements for the original template and is not explicitly prohibited.